md2notion/NotionPyRenderer.py

```diff
@@ -324,3 +331,40 @@ def render_image(self, token):
             'source': token.src,
             'caption': alt
         }
+
+    class __HTMLParser(HTMLParser):
+
+        def __init__(self):
+            super().__init__()
+            self.result = []
+
+        def handle_starttag(self, tag, attrs):
+            if tag != "img": 
+                return
+            src = next((value for key, value in attrs if key == "src"), "")
+            alt = next((value for key, value in attrs if key == "alt"), "No caption")
+            image = {
+                'type': ImageBlock,
+                'display_source': src,
+                'source': src,
+                'caption': alt
+            }
+            self.result.append(image)
+
+    def render_html(self, token):
+        content = token.content
+        parser = self.__HTMLParser()
+        parser.feed(content)
+
+        ret = parser.result
+        ret.insert(0, {
+            'type': TextBlock,
+            'title': content
+        })
+        return ret
+
+    def render_html_block(self, token):
+        return self.render_html(token) 
```

Comment:
Will `token` ever have any `token.children` for these types of blocks?

Reply:
No, there should not be. html blocks and html spans are leaves in the AST tree.